integration_tests/src/main/python/cast_test.py

```diff
@@ -65,6 +65,22 @@ def test_cast_string_date_valid_format():
             lambda spark : unary_op_df(spark, StringGen('[0-9]{1,4}-[0-9]{1,2}-[0-9]{1,2}')).select(f.col('a').cast(DateType())),
             conf = {'spark.rapids.sql.hasExtendedYearValues': 'false'})
 
+@pytest.mark.skipif(is_before_spark_320(), reason="ansi cast(string as date) throws exception only in 3.2.0+")
```

Comment:
Do we need a test to verify we're doing the right thing on Spark versions < 3.2.0, i.e.: either producing a matching result or falling back to the CPU?